webapp/src/ts/app.component.ts

```diff
@@ -416,12 +418,12 @@ export class AppComponent implements OnInit {
   }
 
   private showSessionExpired() {
-    /* Modal({
-     templateUrl: 'templates/modals/session_expired.html',
-     controller: 'SessionExpiredModalCtrl',
-     controllerAs: 'SessionExpiredModalCtrl',
-     singleton: true,
-     });*/
+    if (!this.expiredModalShown) {
```

Comment:
It doesn't look like you're using this property elsewhere, is this so you don't get multiple modals at once? That should be handled by the modalService, if you give your modal an `id`.

Reply:
Yes, it is because of that and it works, in other parts of the code the value is set to true or false, but your approach sounds much better, I'll try, in the Angular 1.x there were a configuration called `singleton` to do this that is not in the current Angular 10 version, but cool that when using an id the modals turns singleton too.

Reply:
yea, I changed the "singleton" thing to work when you give the component an id.